scripts/edit-task.js

```diff
@@ -0,0 +1,230 @@
+import { CLICK_EVENT, HIDE_CLASS } from "/scripts/constants.js";
+import createButton from "/scripts/factory/createButton.js";
+import toggleButtons from "/scripts/factory/toggleButton.js";
+
+function saveEvent(
+  doneButton,
+  editButton,
+  deleteButton,
+  saveAndDoneButton,
+  revertButton,
+  saveButton,
+  inputElement,
+  tasks,
+  id,
+  saveEventHandler,
+  saveAndDoneEventHandler,
+  revertEventHandler
+) {
+  inputElement.contentEditable = false;
+  tasks[id].content = inputElement.textContent;
+  toggleButtons(
+    editButton,
+    deleteButton,
+    revertButton,
+    saveButton,
+    doneButton,
+    saveAndDoneButton
+  );
+  saveButton.removeEventListener(CLICK_EVENT, saveEventHandler);
+  revertButton.removeEventListener(CLICK_EVENT, revertEventHandler);
+  saveAndDoneButton.removeEventListener(CLICK_EVENT, saveAndDoneEventHandler);
```

Comment:
why are we removing the event listeners for the **save**, **revert** etc. buttons inside the edit task handler? are we registering them properly again? 

event handlers should be unregistered before the DOM nodes gets removed.

Reply:
@sam-wtag Yes, When it saves the button, the button gets unmounted, that's why It was needed to unregistered inside the saveEvent handler.